src/language_concepts/generics.md

```diff
@@ -0,0 +1,34 @@
+# Generics
+
+Generics allow you to use the same functions with multiple different concrete data types. You can read more about the concept of generics in the Rust documentation [here](https://doc.rust-lang.org/book/ch10-01-syntax.html).
+
+For example, you may need functions that add `u8` and `u32` integers. Instead of having to write two functions that do the same thing, you can write one function that works for both using generics.
+
+```rust noplaypen
+fn add<T>(a : T, b : T) -> T {
+  return a + b
+}
+```
+
+This concept also works with implementations of structs. For example
+
+```rust noplaypen
+struct BigInt<N> {
+    limbs: [u32; N],
+}
```

Comment:
This is a good example but it is more important for showing generics can be applied to array sizes (which resolve later to some integer) rather than showing a "normal" generic on a struct. Lets add another example of a generic on a struct before this one, and push this one down further and use it as an example of a "numeric generic" on an array size, letting us "be polymorphic over arrays of any length."

The example we could use for structs could be:
```rs
struct RepeatedValue<T> {
    value: T,
    count: comptime Field,
}

impl<T> RepeatedValue<T> {
    fn new(value: T) -> Self {
        Self { value, count: 1 }
    }

    fn increment(mut repeated: Self) -> Self {
        repeated.count += 1;
        repeated
    }

    fn print(self) {
        for _i in 0 .. self.count {
            dep::std::println(self.value);
        }
    }
}

fn main() {
    let mut repeated = RepeatedValue::new("Hello!");
    repeated = repeated.increment();
    repeated.print();
}